pkg/generators/git.go

```diff
@@ -76,20 +77,30 @@ func (g *GitGenerator) generateParamsForGitDirectories(appSetGenerator *argoproj
 }
 
 func (g *GitGenerator) generateParamsForGitFiles(appSetGenerator *argoprojiov1alpha1.ApplicationSetGenerator) ([]map[string]string, error) {
-	allPaths := make(map[string]bool)
+
+	// Get all paths that match the requested path string, removing duplicates
+	allPathsMap := make(map[string]bool)
 	for _, requestedPath := range appSetGenerator.Git.Files {
 		paths, err := g.repos.GetPaths(context.TODO(), appSetGenerator.Git.RepoURL, appSetGenerator.Git.Revision, requestedPath.Path)
 		if err != nil {
 			return nil, err
 		}
 		for _, path := range paths {
-			allPaths[path] = true
+			allPathsMap[path] = true
 		}
 	}
```

Comment:
We can replace this whole block with https://godoc.org/github.com/mpvl/unique#Strings

we can collect all the paths to a list, and then call unique and sort on it

Reply:
My preference is to leave it as is rather than to introduce a new dependency. But if we find we do a lot of this throughout the code, we should reconsider.

Reply:
Tough call.  I'd probably leave it as is.

Reply:
let's leave it for now, and handle it after the milestone.